src/main/kotlin/pl/edu/agh/gem/internal/model/payment/Payment.kt

```diff
@@ -0,0 +1,22 @@
+package pl.edu.agh.gem.internal.model.payment
+
+import java.math.BigDecimal
+import java.time.Instant
+
+data class Payment(
+    val id: String,
+    val groupId: String,
+    val creatorId: String,
+    val recipientId: String,
```

Comment:
i don't see any other  'natural' option to seperate fields apart from Amount add FxData. Think it would only hide the problem of  to many fields and would increase complexity.